src/snowcli/cli/nativeapp/manager.py

```diff
@@ -270,8 +274,20 @@ def _apply_package_scripts(self) -> None:
                 raise InvalidPackageScriptError(relpath, e)
 
         # once we're sure all the templates expanded correctly, execute all of them
-        if self.warehouse:
-            self._execute_query(f"use warehouse {self.warehouse}")
+        try:
+            if self.package_warehouse:
+                self._execute_query(f"use warehouse {self.package_warehouse}")
+            else:
+                raise MissingWarehouseError(
```

Comment:
What about package scripts that don't require a warehouse to excecute? IMO, this should be raised after introspecting the error and seeing it's due to a missing warehouse.

Reply:
Updated.